x-pack/plugins/dataset_quality/server/types.ts

```diff
@@ -6,12 +6,15 @@
  */
 
 import { CustomRequestHandlerContext } from '@kbn/core/server';
+import { FleetSetupContract, FleetStartContract } from '@kbn/fleet-plugin/server';
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface DatasetQualityPluginSetupDependencies {}
+export interface DatasetQualityPluginSetupDependencies {
+  fleet: FleetSetupContract;
```

Comment:
Good question, I already asked fleet team about this one

Reply:
While I haven't get an answer from them I was inspecting kibana code, there are plugin using just the `FleetSetupContract`, although we need to use both because of the way we are registering the plugins.

Reply:
As agreed we will go with this approach and revisit later if a service will suit better our case.